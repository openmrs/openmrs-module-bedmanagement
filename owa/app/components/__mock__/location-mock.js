Object.defineProperties(window.location, {
    host : {
        writable: true,
        value: '192.168.33.10'
    },
    hostname : {
        writable: true,
        value: '192.168.33.10'
    },
    href :{
        writable: true,
        value: 'https://192.168.33.10/openmrs/owa/bedmanagement/admissionLocations.html'
    },
    origin :{
        writable: true,
        value: 'https://192.168.33.10'
    },
    pathname :{
        writable: true,
        value: '/openmrs/owa/bedmanagement/admissionLocations.html'
    },
    port :{
        writable: true,
        value: ''
    },
    protocol :{
        writable: true,
        value: 'https:'
    }
});