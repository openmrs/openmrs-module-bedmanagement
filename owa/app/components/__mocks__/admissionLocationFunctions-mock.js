import pickBy from 'lodash.pickby';
import {admissionLocations, visitLocations, bedTypes} from 'utilities/__tests__/testData.json';

const admissionLocationFunctionsMock = {
    setActiveLocationUuid: (admissionLocationUuid) => jest.fn(),
    getActiveLocationUuid: () => {
        return 'baf7bd38-d225-11e4-9c67-080027b662ec';
    },
    setState: (data) => jest.fn(),
    getState: () => jest.fn(),
    getAdmissionLocations: () => {
        return admissionLocations;
    },
    getVisitLocations: () => {
        return visitLocations;
    },
    getBedTypes: () => {
        return bedTypes;
    },
    getAdmissionLocationByUuid: (admissionLocationUuid) => {
        return typeof admissionLocations[admissionLocationUuid] !== 'undefined' ?
            admissionLocations[admissionLocationUuid] : null;
    },
    getParentAdmissionLocation: (admissionLocationUuid) => {
        const parentAdmissionLocationUuid = admissionLocations[admissionLocationUuid].parentAdmissionLocationUuid;
        return parentAdmissionLocationUuid != null ? admissionLocations[parentAdmissionLocationUuid] : null;
    },
    getChildAdmissionLocations : (admissionLocationUuid) => {
        return  pickBy(admissionLocations, (admissionLocation) => {
            return admissionLocation.parentAdmissionLocationUuid == admissionLocationUuid;
        });
    },
    reFetchAllAdmissionLocations : () => jest.fn(),
    notify : (notifyType, message) => jest.fn()
};

export default admissionLocationFunctionsMock;
