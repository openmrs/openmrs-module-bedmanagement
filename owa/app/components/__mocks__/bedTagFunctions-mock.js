import _ from 'lodash';

import {bedTags} from 'utilities/__tests__/testData.json';

const bedTagFunctionsMock = {
    setState: (stateData) =>  jest.fn(),
    getBedTags: () => {
        return bedTags;
    },
    getBedTagByUuid: (bedTagUuid) => {
        return _.find(bedTags, function (bedTag) {
            return bedTag.uuid == bedTagUuid;
        });
    },
    fetchBedTags: () => jest.fn(),
    notify : (notifyType, message) => jest.fn()
};

export default bedTagFunctionsMock;
