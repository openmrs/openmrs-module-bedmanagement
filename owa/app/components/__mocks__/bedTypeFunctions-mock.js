import _ from 'lodash';

import {bedTypes} from 'utilities/__tests__/testData.json';

const bedTypeFunctionsMock = {
    setState: (stateData) =>  jest.fn(),
    getBedTypes: () => {
        return bedTypes;
    },
    getBedTypeById: (bedTypeId) => {
        return _.find(bedTypes, function (bedType) {
            return bedType.id == bedTypeId;
        });
    },
    getBedTypeName: (bedTypeName) => {
        return _.find(bedTypes, function (bedType) {
            return bedType.name == bedTypeName;
        });
    },
    fetchBedTypes: () => {
        return bedTypes;
    },
    notify : (notifyType, message) => jest.fn()
};

export default bedTypeFunctionsMock;
