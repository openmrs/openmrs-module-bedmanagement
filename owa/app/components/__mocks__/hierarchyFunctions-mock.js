import {admissionLocations} from 'utilities/__tests__/testData.json';

const hierarchyFunctionMock = {
    getActiveUuid: () => {
        return 'baf7bd38-d225-11e4-9c67-080027b662ec';
    },
    getAdmissionLocations: () => {
        return admissionLocations;
    },
    setActiveUuid: (locationUuid) => jest.fn(),
    setState: (stateData) => jest.fn(),
    toggleIsOpen: (locationUuid, isOpen) => jest.fn(),
    setAdmissionLocationIsOpen: (locationUuid, isOpen) => jest.fn()
};

export default hierarchyFunctionMock;