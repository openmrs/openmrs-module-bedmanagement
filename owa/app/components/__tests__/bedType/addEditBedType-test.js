import React from 'react';
import {shallow, mount} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

import AddEditBedType from 'components/bedType/bedTypeForm/addEditBedType';
import bedTypeFunctionsMock from 'components/__mocks__/bedTypeFunctions-mock';

const testProps = {
    bedTypeFunctions: bedTypeFunctionsMock
};

require('components/__mocks__/location-mock');
describe('AddEditBedType', () => {
    beforeAll(() => {
        var mock = new MockAdapter(axios);
        const data = {
            'id':2,
            'name':'luxury',
            'displayName': 'luxury bed',
            'description': 'LXY'
        };

        mock.onPost('https://192.168.33.10/openmrs/ws/rest/v1/bedtype/2')
            .reply(200, data);
    });

    it('Should display add edit bed type form properly', () => {
        const addBedTypeForm = shallow(<AddEditBedType bedTypeFunctions={testProps.bedTypeFunctions} bedTypeId={null} operation='add'/>);
        expect(addBedTypeForm.find('#name-field').props().value).toBe('');
        expect(shallowToJson(addBedTypeForm)).toMatchSnapshot();


        const editBedTypeForm = shallow(<AddEditBedType bedTypeFunctions={testProps.bedTypeFunctions} operation='edit'
            bedTypeId={2} />);

        expect(editBedTypeForm.find('#name-field').props().value).toBe('luxury');
        expect(editBedTypeForm.find('#display-name-field').props().value).toBe('luxury bed');
        expect(editBedTypeForm.find('#description-field').props().value).toBe('LXY');
        expect(shallowToJson(editBedTypeForm)).toMatchSnapshot();
    });

    it('Should trigger event handler', () => {
        const spyOnCancelEventHandler = jest.spyOn(AddEditBedType.prototype, 'cancelEventHandler');
        const spyOnSubmitHandler = jest.spyOn(AddEditBedType.prototype, 'onSubmitHandler');
        const spyOnChangeNameField = jest.spyOn(AddEditBedType.prototype, 'onChangeNameField');
        const spyOnChangeDisplayNameField = jest.spyOn(AddEditBedType.prototype, 'onChangeDisplayNameField');
        const spyOnChangeDescription = jest.spyOn(AddEditBedType.prototype, 'onChangeDescription');
        const spySetState = jest.spyOn(testProps.bedTypeFunctions, 'setState');
        const editBedTypeForm = mount(<AddEditBedType bedTypeFunctions={testProps.bedTypeFunctions} operation='edit' bedTypeId={2}/>);

        editBedTypeForm.find('#name-field').simulate('change');
        expect(spyOnChangeNameField).toHaveBeenCalled();
        editBedTypeForm.find('#display-name-field').simulate('change');
        expect(spyOnChangeDisplayNameField).toHaveBeenCalled();
        editBedTypeForm.find('#description-field').simulate('change');
        expect(spyOnChangeDescription).toHaveBeenCalled();

        editBedTypeForm.find('input[name=\'cancel\']').simulate('click');
        expect(spyOnCancelEventHandler).toHaveBeenCalled();
        expect(spySetState).toHaveBeenCalledWith({
            activePage: 'listing',
            pageData: {}
        });

        editBedTypeForm.find('input[type=\'submit\']').simulate('submit');
        expect(spyOnSubmitHandler).toHaveBeenCalled();
    });
});