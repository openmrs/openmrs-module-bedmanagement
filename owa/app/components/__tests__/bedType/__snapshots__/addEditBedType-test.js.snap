// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddEditBedType Should display add edit bed type form properly 1`] = `
<div
  className="form-container"
>
  <div
    className="bed-type-form"
  >
    <div
      className="block"
    >
      <div
        className="block-title"
      >
        Add
         Bed Type
        <a
          className="back-link"
          href="javascript:void(0);"
          onClick={[Function]}
          title="back"
        >
          &lt;&lt; Back
        </a>
      </div>
      <div
        className="block-content"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-block"
          >
            <label
              className="form-title"
            >
              Name
            </label>
            <input
              id="name-field"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
          </div>
          <div
            className="form-block"
          >
            <label
              className="form-title"
            >
              Display Name
            </label>
            <input
              id="display-name-field"
              onChange={[Function]}
              required={true}
              type="text"
              value=""
            />
          </div>
          <div
            className="form-block"
          >
            <label
              className="form-title"
            >
              Description
            </label>
            <textarea
              id="description-field"
              onChange={[Function]}
              value=""
            />
          </div>
          <div
            className="form-block"
          >
            <input
              className="btn btn-primary float-left margin-right"
              disabled={false}
              name="submit"
              type="submit"
              value="Save"
            />
            <input
              className="btn btn-danger float-left"
              name="submit"
              onClick={[Function]}
              type="button"
              value="Cancel"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`AddEditBedType Should display add edit bed type form properly 2`] = `
<div
  className="form-container"
>
  <div
    className="bed-type-form"
  >
    <div
      className="block"
    >
      <div
        className="block-title"
      >
        Edit
         Bed Type
        <a
          className="back-link"
          href="javascript:void(0);"
          onClick={[Function]}
          title="back"
        >
          &lt;&lt; Back
        </a>
      </div>
      <div
        className="block-content"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-block"
          >
            <label
              className="form-title"
            >
              Name
            </label>
            <input
              id="name-field"
              onChange={[Function]}
              required={true}
              type="text"
              value="luxury"
            />
          </div>
          <div
            className="form-block"
          >
            <label
              className="form-title"
            >
              Display Name
            </label>
            <input
              id="display-name-field"
              onChange={[Function]}
              required={true}
              type="text"
              value="luxury bed"
            />
          </div>
          <div
            className="form-block"
          >
            <label
              className="form-title"
            >
              Description
            </label>
            <textarea
              id="description-field"
              onChange={[Function]}
              value="LXY"
            />
          </div>
          <div
            className="form-block"
          >
            <input
              className="btn btn-primary float-left margin-right"
              disabled={false}
              name="submit"
              type="submit"
              value="Save"
            />
            <input
              className="btn btn-danger float-left"
              name="submit"
              onClick={[Function]}
              type="button"
              value="Cancel"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
